### DISCUSSION TITLE
Function of second UDP port

### DISCUSSION THREAD

**Discussion by bennypi** [infinispan/infinispan#12662] [Q&A]
Hi everyone, 

we are using infinispan embedded in a spring boot application with the TCP stack. I checked which ports the application listens on and saw this result:

- UDP: 46655
- UDP: 46384 (random high port)
- TCP: 7800
- TCP: 57800

I think I understand three out of the four ports:
- UDP 46655 is used for UDP Multicast discovery, as per https://infinispan.org/docs/stable/titles/embedding/embedding.html#jgroups-ports_cluster-transport
- TCP 7800 is used for TCP transport, as per https://infinispan.org/docs/stable/titles/embedding/embedding.html#jgroups-ports_cluster-transport
- TCP 57800 is used for failure discovery FD_SOCK2, as per `default-jgroups-tcp.xml` and the log entry `o.j.p.FD_SOCK2: server listening on *.57800`

However, I am having a hard time understanding what the function of the second UDP port is. Additionally, it would be nice to know if this port can be set to a static value, so that admins can specify a firewall rule for this port if necessary.

**Comment by bennypi** [comment]
After some experiments with wireshark, it seems that the second UDP port is used to send the IP Multicast messages, which are received on the first UDP port, so to me it looks like this is an outbound port. I am wondering, is this the default behavior for outbound UDP ports that they are also listening on the port?

**Comment by tristantarrant** [comment]
@belaban
